providers/common/include/internal/providercommon.h

```diff
@@ -8,7 +8,58 @@
  */
 
 #include <openssl/provider.h>
+#include <openssl/engine.h>
 
 const OSSL_PROVIDER *FIPS_get_provider(OPENSSL_CTX *ctx);
 
 const char *ossl_prov_util_nid_to_name(int nid);
+
+typedef struct {
+    /*
+     * References to the underlying cipher implementation.  cipher caches
+     * the cipher, always.  alloc_cipher only holds a reference to an explicitly
+     * fetched cipher.
+     * cipher is cleared after an init call.
+     */
+    const EVP_CIPHER *cipher;   /* cipher */
+    EVP_CIPHER *alloc_cipher;   /* fetched cipher */
+
+    /*
+     * Conditions for legacy EVP_MD uses.
+     * tmpengine is cleared after an init call.
+     */
+    ENGINE *engine;             /* cipher engine */
+} PROV_CIPHER;
+
+typedef struct {
+    /*
+     * References to the underlying digest implementation.  tmd caches
```

Comment:
tmd

Reply:
You are out of date -- I split these definitions into a separate file.
The only thing in `providercommon.h` isn't suitable for FIPS.

(BTW: I think it should be called `provider_common.h` and nothing it contains is _common_)